---
title: "DBA QC panel R Notebook"
output: html_notebook
---

This is an [R Markdown](http://rmarkdown.rstudio.com) Notebook. When you execute code within the notebook, the results appear beneath the code. 

Try executing this chunk by clicking the *Run* button within the chunk or by placing your cursor inside it and pressing *Cmd+Shift+Enter*. 

#Set work path, propigate through all chunk
```{r setup}
knitr::opts_knit$set(root.dir = normalizePath("/Users/graham/bioinf/dba/amplicon_qc/lanes_17_20"))
#setwd(dir = "/Users/graham/bioinf/dba/amplicon_qc/lanes_7_15")
```
# Single library analysis
```{r}
library("TarSeqQC")
library("BiocParallel")
```
# Queue up the files paths
```{r}
# Run 7 bam example - TODO: use a median example for in depth qc analysis
bamFile <- list.files(path="dataset/R_2017_09_27_16_11_46_user_SN1-487-DBAv6_L19", pattern="IonXpress_001.*.bam$", full.names = TRUE)

fastaFile = "../hg19.fa"

# bedFile
bedFile = "dataset/DBAv6.formatted.bed"

```
# Check essential files. BED and BAM.
```{r}
checkBedFasta(bedFile = bedFile, fastaFile = fastaFile)
```
# Constructor call
```{r}
BPPARAM<-bpparam()
myPanel<-TargetExperiment(bedFile, bamFile, fastaFile, feature="amplicon", attribute="coverage", BPPARAM=BPPARAM)

setFeature(myPanel)<-"amplicon"
setAttribute(myPanel)<-"coverage"

# definition of the interval extreme values
attributeThres<-c(0,1,30,100,Inf)

```
# Read freq per low coverage amplicon
```{r}
readFrequencies(myPanel)

p <- plotInOutFeatures(readFrequencies(myPanel))

p <-plotGeneAttrPerFeat(myPanel, geneID="RPS5") +
  theme(axis.title=element_text(size=12), title= element_text(size=12), axis.text.x = element_text(size=10))
  
p
```
# Plot actual bam coverage across gene regions
```{r}
getRegion(myPanel, level="gene", ID="RPS19", collapse=FALSE)

g<-plotFeature(myPanel, featureID="AMPL3707075462")
g

# plot RPS19
getRegion(myPanel, level="gene", ID="RPS19", collapse=FALSE)

g1 <- plotRegion(myPanel, region=c(42364769,42375473), seqname="chr19", SNPs=TRUE, minAAF = 0.05, minRD = 0, xlab="", title="RPS19 (chr19) coverage by 5 amplicons", size=1) +
  annotate("text", x = 42365200, y = -3, label = "AMPL1501181774") +
  annotate("text", x = 42365300, y = -7, label = "AMPL886902177") +
  annotate("text", x = 42373100, y = -3, label = "AMPL5322931146") +
  annotate("text", x = 42373700, y = -7, label = "AMPL5322938416") +
  annotate("text", x = 42375100, y = -3, label = "AMPL3707075462") +
  xlab("Chromosome position")

g1# focus on RPS19 failing AMPL3707075462 amplicon
g1.1 <- plotRegion(myPanel, region=c(42375349,42375473), seqname="chr19", SNPs=TRUE, minAAF = 0.05, minRD = 0, xlab="", title="RPS19 (chr19) coverage by amplicon AMPL3707075462", size=1) +
  xlab("Chromosome position")

# plot RPS5
getRegion(myPanel, level="gene", ID="RPS5", collapse=FALSE)

g2 <- plotRegion(myPanel, region=c(58899450,58906199), seqname="chr19", SNPs=TRUE, minAAF = 0.05, minRD = 0, xlab="", title="RPS5 (chr19) coverage by 6 amplicons", size=1) +
  annotate("text", x = 58899700, y = -3, label = "AMPL5323842280") +
  annotate("text", x = 58904300, y = -3, label = "AMPL882331318") +
  annotate("text", x = 58904400, y = -7, label = "AMPL882358879") +
  annotate("text", x = 58904800, y = -15, label = "AMPL4849070270") +
  annotate("text", x = 58905800, y = -3, label = "AMPL5323581504") +
  annotate("text", x = 58905900, y = -7, label = "AMPL5323582863") +
  xlab("Chromosome position")

# plot RPL36A
getRegion(myPanel, level="gene", ID="RPL36A", collapse=FALSE)

g3 <- plotRegion(myPanel, region=c(100645882,100650832), seqname="chrX", SNPs=TRUE, minAAF = 0.05, minRD = 0, xlab="", title="RPL36A (chrX) coverage by 6 amplicons", size=1) +
  annotate("text", x = 100646050, y = -3, label = "AMPL883962734") +
  annotate("text", x = 100646500, y = -5, label = "AMPL5323738783") +
  annotate("text", x = 100646800, y = -7, label = "AMPL5323069303") +
  annotate("text", x = 100650200, y = -3, label = "AMPL882373081") +
  annotate("text", x = 100650550, y = -5, label = "AMPL882376993") +
  annotate("text", x = 100650600, y = -7, label = "AMPL5323077738") +
  xlab("Chromosome position")

```

